### PR TITLE
Consolidate coordinator SSE lifecycle and completion policies

### PR DESCRIPTION
**Pull request by Gajesh2007**
The coordinator maintained three copies of the provider SSE relay loop and two copies of translated-endpoint setup, error settlement and completion handling. This change gives channel arbitration one implementation and makes endpoint completion differences explicit, so changes to coalescing or late-error delivery have one owner.

Stacked on `feat/gpt-oss-20b-prefix-cache` (#894); no release/default changes are introduced here.

- Production streaming code shrinks by 127 lines; regression tests and documentation account for the overall diff growth.
- `relayProviderStream` owns bounded queued-chunk draining, flush-before-terminal ordering, idle timeout and cancellation selection for Chat, Responses, Completions and Messages.
- `handleEndpointStreamingResponse` owns shared translated-endpoint setup, initial provider errors, route outcomes, refunds and final flushing. Emitters retain their exact wire formats; chat retains its held finish/usage/metadata logic.
- `streamCompletionPolicy` explicitly preserves existing differences: Messages/Completions require completion usage even without a reservation and observe cancellation during the completion wait; Responses retains its settled-reservation fallback and independent usage grace period.
- The consumer architecture page documents those ownership boundaries and policies.

Before:

```mermaid
flowchart TD
  A[Committed provider request] --> B{Consumer endpoint}
  B --> C[Chat handler: own select/drain/flush loop]
  B --> D[Responses handler: own setup/error/select loop]
  B --> E[Generic handler: own setup/error/select loop]
  C --> F[Chat held usage/finish and metadata]
  D --> G[Responses completion grace and emitter]
  E --> H[Explicit completion required and Messages/Completions emitters]
  C --> I[Queued chunks before error; refund/route outcome]
  D --> I
  E --> I
  F --> J[Existing endpoint success terminal]
  G --> J
  H --> J
  I --> K[Existing endpoint in-band error or client departure]
```

After:

```mermaid
flowchart TD
  A[Committed provider request] --> B{Consumer endpoint}
  B --> C[Chat handler: chat wire state]
  B --> D[handleEndpointStreamingResponse: shared setup/error settlement]
  C --> E[relayProviderStream: one select/drain/flush loop]
  D --> E
  E -->|Closed| F[Chat held usage/finish and metadata]
  E -->|Closed| G[finishEndpointStream: explicit completion policy]
  G --> H[Responses or Messages/Completions emitter]
  F --> J[Same endpoint success terminal]
  H --> J
  E -->|Error / idle timeout / client gone| I[Caller settles and records outcome]
  I --> K[Same endpoint in-band error or client departure]
```

Validation:

- Existing SSE golden, burst, coalescing, usage and Responses tests pass.
- New regression matrix covers missing completion across all translated formats with no/outstanding/settled reservations, exactly-once refunds, dispatch-time chunks before an initial error, and the endpoint-specific cancellation grace.
- `GOTOOLCHAIN=go1.25.0 go test ./coordinator/...` passes (API package 194.418s; registry/routing simulation/store/billing and all other coordinator packages pass).
- Final targeted API race suite passes (47.427s), covering endpoint, SSE/burst/coalescing, settlement, generic and request-outcome tests.
- `git diff --check` and docs lint pass (278 documents).

The observable API shapes, status codes, flush bounds, timer-reset placement, settlement calls, telemetry, provider cancellation ownership and GPT-OSS defaults remain unchanged. This is a focused coordinator refactor, not a rewrite of the scheduler or ledger.

### CODE REVIEW

**Comment by Gajesh2007** [comment]
@codex review

File: coordinator/api/endpoint_stream.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		s.refundReservedBalance(pr, "provider_timeout:"+pr.RequestID)
+		s.ddIncr("inference.in_band_error", []string{"model:" + pr.Model, "reason:timeout"})
+		s.updateInferenceRouteOutcomeForPending(pr, postCommitStreamTimeoutOutcome(pr))
+		emitter.emitError("timeout", "request timed out")
+	case providerStreamClientGone:
+		profileClientGone(pr, phaseAfterCommit)
+	}
+}
+
+func newEndpointStreamEmitter(w http.ResponseWriter, flusher http.Flusher, pr *registry.PendingRequest) (endpointStreamEmitter, streamCompletionPolicy) {
```
[the remark this answers is not in the document]
Comment (Gajesh2007) [reply coordinator/api/endpoint_stream.go:104]: Fixed in 77f0ea83b5be6e1d7c89ad21308d5a5a6300f4cf. The canonical API table now cites newEndpointStreamEmitter in coordinator/api/endpoint_stream.go and has a refreshed stamp. A repository-wide search of current Markdown (including design and root docs, excluding frozen reports/releases) found no remaining references to the removed stream symbols. Docs lint passes (278 files).

**Comment by Gajesh2007** [comment]
@codex review

**Comment by Gajesh2007** [comment]
Final validation for `77f0ea83b5be6e1d7c89ad21308d5a5a6300f4cf`:

- All ordinary PR CI checks, including provider and E2E integration tests, pass. The separate environment-gated E2E benchmark job remains waiting.
- Codex completed review of this head with a thumbs-up; no unresolved review threads remain.
- The five sibling refactors (#897, #898, #899, #900, #901) combine cleanly in a local validation branch based on GPT-OSS #894. Combined coordinator streaming/settlement race tests, testbed and targeted E2E CPU race tests pass. The final combination also passes all 73 benchmark-wrapper tests, docs lint (279 files), and whitespace checks.

Individual validation details and limitations are in each PR description, including the dedicated-M5 provider/live GPT-OSS checks and the unchanged baseline TypeScript diagnostics in #899. No merge or deployment was performed.

**Comment by Gajesh2007** [comment]
@codex review

**Comment by Gajesh2007** [comment]
@codex review

**Comment by Gajesh2007** [comment]
Superseded by #1054, which carries this change and its regression coverage into the coordinator subsystem layout. The replacement passed the relevant local race tests, build/lint checks, and independent source review.

Please review #1054; closing this older implementation to keep one active version of the change. The branch and review history are retained. No change has been merged to master by this closure.
